stdlib/map.mli

```diff
@@ -86,6 +86,16 @@ module type S =
        of [x] in [m] disappears.
        @before 4.03 Physical equality was not ensured. *)
 
+    val update: key -> ('a option -> 'a) -> 'a t -> 'a t
```

Comment:
This is, in fact, already done by @sbriais's implementation. But it could also be useful to specify it in the interface documentation.